src/uk/gov/hmcts/cmc/integrationtests/IntegrationTests.groovy

```diff
@@ -83,41 +83,44 @@ class IntegrationTests implements Serializable {
     steps.sh "${dockerComposeCommand()} pull"
   }
 
-  private void startTestEnvironment(String application) {
+  private void startTestEnvironment() {
     steps.sh """
               ${dockerComposeCommand()} up --no-color -d remote-webdriver \\
-                         ${application}          
+                         citizen-frontend \\
+                         legal-frontend
               """
   }
 
-  private void executeTests(Team team) {
-    steps.sh """
+  private void executeTests() {
+    def exitCode = steps.sh returnStatus: true, script: """
             mkdir -p output
-            ${dockerComposeCommand()} up --no-deps --no-color ${team.testsContainerName}
+            ${dockerComposeCommand()} ${runCommand()}
             """
-    analyseTestResults(team)
+    archiveDockerLogs()
+
+    if (exitCode > 0) {
+      steps.archiveArtifacts 'output/*.png'
+      steps.error("Integration tests failed")
+    }
+  }
+
+  private String runCommand() {
+    String testsTag = env.TESTS_TAG
+    if (testsTag == null || testsTag.trim().isEmpty()) {
+      return "run --no-deps integration-tests"
+    } else {
+      return "run --no-deps integration-tests test -- --grep '${testsTag}'"
+    }
   }
 
-  private void executeCrossBrowserTests(Team team) {
+  private void executeCrossBrowserTests() {
     steps.sh """
             mkdir -p output
             ${dockerComposeCommand()} up --no-color -d saucelabs-connect
             ./bin/run-cross-browser-tests.sh
             """
   }
 
-  private void analyseTestResults(Team team) {
-    def testExitCode = steps.sh returnStdout: true,
-      script: "${dockerComposeCommand()} ps -q ${team.testsContainerName} | xargs docker inspect -f '{{ .State.ExitCode }}'"
```

Comment:
Don't we need `.State.ExitCode` anymore? Is run command returning exit code properly?

Reply:
We don't, `run` will return the exit code of command it executed.

Reply:
☝️ it's not documented, it's something that you need to find out by yourself 😂

Reply:
Yeah, expected run to do it. We introduced `State.ExitCode` because we had `docker-compose up` initially.